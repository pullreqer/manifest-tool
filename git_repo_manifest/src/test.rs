use crate::Manifest;
use insta::{assert_debug_snapshot, glob};
use quick_xml::de::{from_str, DeError};
use std::fs;
#[test]
fn run_test_inputs() {
    glob!("test_inputs/*.xml", |path| {
        let input = fs::read_to_string(path).unwrap();
        let foo: Result<Manifest, DeError> = from_str(&input);

        assert_debug_snapshot!(foo);
    });
}
