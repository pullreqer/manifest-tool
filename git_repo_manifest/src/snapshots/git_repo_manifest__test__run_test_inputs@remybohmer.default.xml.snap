---
source: git_repo_manifest/src/test.rs
expression: foo
input_file: git_repo_manifest/src/test_inputs/remybohmer.default.xml
---
Ok(
    Manifest {
        notice: Some(
            Notice {
                notice: Some(
                    "You checked out the default manifest",
                ),
            },
        ),
        manifest_server: None,
        remotes: [],
        default: None,
        remove_projects: [],
        projects: [
            Project {
                name: "remybohmer/demo-project-1",
                path: Some(
                    "project1",
                ),
                remote: None,
                revision: None,
                dest_branch: None,
                groups: None,
                rebase: None,
                sync_c: None,
                sync_s: None,
                sync_tags: None,
                upstream: None,
                clone_depth: None,
                force_path: None,
            },
            Project {
                name: "remybohmer/demo-project-2",
                path: Some(
                    "project2",
                ),
                remote: None,
                revision: None,
                dest_branch: None,
                groups: None,
                rebase: None,
                sync_c: None,
                sync_s: None,
                sync_tags: None,
                upstream: None,
                clone_depth: None,
                force_path: None,
            },
            Project {
                name: "dpursehouse/lfs-test",
                path: Some(
                    "lfs-test",
                ),
                remote: None,
                revision: None,
                dest_branch: None,
                groups: None,
                rebase: None,
                sync_c: None,
                sync_s: None,
                sync_tags: None,
                upstream: None,
                clone_depth: None,
                force_path: None,
            },
            Project {
                name: "remybohmer/demo-repo-hooks",
                path: Some(
                    ".repo_hooks",
                ),
                remote: None,
                revision: None,
                dest_branch: None,
                groups: None,
                rebase: None,
                sync_c: None,
                sync_s: None,
                sync_tags: None,
                upstream: None,
                clone_depth: None,
                force_path: None,
            },
        ],
        extend_projects: [],
        repo_hooks: Some(
            RepoHooks {
                in_project: "remybohmer/demo-repo-hooks",
                enabled_list: [
                    "post-sync",
                    "pre-upload",
                ],
            },
        ),
        includes: [
            Include {
                name: "common/server_settings.xml",
            },
        ],
    },
)
